rpp30-boc-heartbleed/whimsical
# Shared components

## Description

Hello, this PR will only be for the shared components & modal component I created. 
I also created sample icons if you guys need to use them to test them in your components for MVP.

## How to test

To use the icons, import icons from the constants folder I created into your components with: 
<b>import { COLORS, SIZES, icons } from '../../constants'; </b>

* I imported the assets into a new constants folder and consolidated all .png files into the constants folder for easier importing/sharing.
* I re-installed react-native-paper into the <b>client folder package.json </b>
* I tested all of the shared components in my 'ErrandMap' component to make sure they didn't break the app, and were functioning as expected.
* I created a .gitignore for the client folder as well, decided to use <b>.env</b> for my Google API key. Unsure if having a separate <b>.env</b> on client side to store credentials would be a security breach since server-side is more secure but mehhh. Will look into Firebase more after MVP.
<b>.gitignored files: </b> 
- node_modules/. 
- .env*
* <b>Dependencies Installed: </b>
   - react-native-paper: "^4.11.1"
   - react-native-animation-library: "0.0.8"
   - react-native-dotenv: "^3.3.1"
   - @react-native-community/geolocation: "^2.0.2"
   - react-native-maps-directions: "^1.8.0"

***Still getting used to recoil, so used useState for now.
   
<b> Please only merge this PR after Aaron's, if approved. Thank you!</b>


## Notes

I commented out my ErrandTrackerContainer component - still am working on it and will PR later this weekend on it.

## Issue tracking
https://trello.com/c/ZY4Bv1j1/38-shared-components

File Structure
* Created an icons folder under assets folder and moved icons there
* Created a constants folder

![image](https://user-images.githubusercontent.com/56424589/147846450-459faa0a-9319-4ae0-b289-ade3cacdb5a7.png)


Screenshot of icon usage of my *currently* not finished poopy component. I will use mock data for MVP on Monday.

![image](https://user-images.githubusercontent.com/56424589/147846205-8d28d2f6-f232-4623-a8fd-56e14325ee06.png)

A big mess of imports *example*
![image](https://user-images.githubusercontent.com/56424589/147846222-69bea6d0-7b47-4523-a4f5-db8981ef451c.png)

How I used the icon in my code after importing into my component
![image](https://user-images.githubusercontent.com/56424589/147846317-aab0c5e1-5bbb-49ad-8fba-77d349a6ea58.png)
